trade-finance-manager-ui/server/controllers/helpers/amendments.helper.show-amendment-button.test.js

```diff
@@ -0,0 +1,90 @@
+import { TEAM_IDS, DEAL_STATUS } from '@ukef/dtfs2-common';
+import { showAmendmentButton } from './amendments.helper';
+import { AmendmentButtonDealBuilder } from '../../test-mocks/show-amendments-button-deal-builder';
+import { DEAL } from '../../constants';
+
+describe('amendments helper', () => {
+  describe('showAmendmentButton()', () => {
+    const defaultUserTeams = [TEAM_IDS.PIM];
+
+    it.each(getSuccessTestCases())('should return true when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(true);
+    });
+
+    it.each(getFailureTestCases())('should return false when $description', ({ deal, userTeams }) => {
+      const result = showAmendmentButton(deal, userTeams);
+      expect(result).toEqual(false);
+    });
+
+    function getSuccessTestCases() {
+      return [
+        {
+          description: 'submission type is AIN',
+          deal: new AmendmentButtonDealBuilder().withDealSnapshotSubmissionType(DEAL.SUBMISSION_TYPE.AIN).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'submissionType is MIN',
+          deal: new AmendmentButtonDealBuilder().withDealSnapshotSubmissionType(DEAL.SUBMISSION_TYPE.MIN).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'deal stage is confirmed',
+          deal: new AmendmentButtonDealBuilder().withTfmStage(DEAL.DEAL_STAGE.CONFIRMED).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'deal stage is amendment in progress',
+          deal: new AmendmentButtonDealBuilder().withTfmStage(DEAL.DEAL_STAGE.AMENDMENT_IN_PROGRESS).build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'user is in PIM team',
+          deal: new AmendmentButtonDealBuilder().build(),
+          userTeams: defaultUserTeams,
+        },
+        {
+          description: 'user is in PIM team and another team',
+          deal: new AmendmentButtonDealBuilder().build(),
+          userTeams: [...defaultUserTeams, TEAM_IDS.UNDERWRITER_MANAGERS],
+        },
+        {
+          description: 'deal is not cancelled or pending cancellation',
+          deal: new AmendmentButtonDealBuilder().withStatus(DEAL_STATUS.COMPLETED).build(),
+          userTeams: defaultUserTeams,
+        },
+      ];
+    }
```

Comment:
For these success cases, do we not need multiple of the criteria to be true? I'm wondering if it makes more sense to structure these tests as
```
describe('when the user is logged in as PIM',() => {
  describe.each(AIN, MIN)('when the submission type is %s', () => {
    // etc ...
  })
})
```

I love the builder - would it be possible to still use it with a structure like that? I think the failure tests make sense to stay as is (since any one failure means its all a failure right?)

Reply:
Builders usually build a valid object for a situation -- the only reason to include these is to expressively check individual parameters... we could rework this, but I think it adds extra complexity and maintainability issues. Let me give it a go, and we can revert if it looks bad